base/error.jl

```diff
@@ -218,6 +219,9 @@ julia> @assert isodd(3) "What even are numbers?"
 ```
 """
 macro assert(ex, msgs...)
+    enable_asserts = !isdefined(Main, :Base) || Main.Base.julia_debug
```

Comment:
`!isdefined(Main, :Base)`

I assume this is the canonical way to check if something is running as a script? possibly should factor that out into an independent `iscurrentscript` function somewhere

Reply:
This is not checking if it is a script actually, it is making sure that asserts are emitted when e.g. building the compiler (and `Base` is not defined).

Reply:
gotcha